### PR TITLE
refactoring infographics_image_generator and adding upload to s3

### PR DESCRIPTION
**Pull request by tkalir**

### CODE REVIEW

File: anyway/parsers/cbs/s3/base.py [hunk 1]
```diff
@@ -4,11 +4,11 @@
 
 
 class S3DataClass:
-    def __init__(self):
+    def __init__(self, s3_bucket_name=None):
         self._aws_access_key = secrets.get("AWS_ACCESS_KEY")
         self._aws_secret_key = secrets.get("AWS_SECRET_KEY")
         self._s3_resource = None
-        self._s3_bucket = None
+        self._s3_bucket = self.s3_resource.Bucket(s3_bucket_name) if s3_bucket_name else None
```
Comment (ziv17) [inline anyway/parsers/cbs/s3/base.py:11]: If `s3_bucket_name` evaluates to `True`, then this line will try to access an attribute of `None`?

File: anyway/infographic_image_generator.py [hunk 2]
```diff
@@ -34,53 +41,90 @@ def get_download_button_elements(browser):
...
         time.sleep(1)
-        files = os.listdir(download_directory)
-        if len(files) == number_of_expected_files:
-            return True
-    return False
+        contents = requests.get(
+            f"{selenium_remote_results_url}/{newsflash_id}/").json()
+        filenames = [item['name'] for item in contents]
+        unique_filenames = get_unique_filenames(filenames)
+        if len(unique_filenames) == number_of_expected_files:
```
Comment (ziv17) [inline anyway/infographic_image_generator.py:67]: I think there is a remote possibility that this function will terminate after the last file was created, but before it was completely downloaded.

**Review by ziv17 — APPROVED** [review]
Well done!

One comment below

File: anyway/infographic_image_generator.py (hunk 2 above)
Comment (tkalir) [reply anyway/infographic_image_generator.py:67]: yes, well, this function is supposed to try and wait until everything is downloaded, the question is what the timeout should be. usually 1 is enough, but we can make it let's say 60 seconds to be sure. what do you think?

Comment (tkalir) [reply anyway/infographic_image_generator.py]: Ok, I see your point. On the discussion below they say that Chrome uses an extension for partial files that were not completely downloaded yet, so I added a check that no file has this extension.

https://stackoverflow.com/questions/34338897/python-selenium-find-out-when-a-download-has-completed

Comment (atalyaalon) [reply anyway/infographic_image_generator.py]: @ziv17 @tkalir 
Great work.
I approved and merged. However if you think this issue should be further discussed, let's open a new issue.

**Pull request merged**